### PR TITLE
Fix java.lang.IllegalStateException: The bundle inputstream:aQute.xra…

### PR DESCRIPTION
**Pull request by adimova**
…y.plugin-1.0.13.jar is already uninstalled.

Signed-off-by: Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java [hunk 1]
```diff
@@ -288,19 +288,27 @@ protected void processAutomationProvider(Bundle bundle) {
...
                 }
             }
         }
-        Enumeration<URL> urlEnum = bundle.findEntries(path, null, false);
-        if (urlEnum == null)
+        Enumeration<URL> urlEnum = null;
+        try {
+            urlEnum = bundle.findEntries(path, null, false);
+        } catch (IllegalStateException e) {
+            logger.error("Can't read from resource of bundle with ID " + bundle.getBundleId(), e);
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java]: Is this really an error situation? I thought that this exception came up on a regular shutdown of the OSGi framework - in such a situation, we should not write error messages to the log.

**Comment by kaikreuzer** [comment]
lgtm. What makes me a bit worried though is that you did a lot of other fixes in this PR and are now often doing checks for the inverse of what has been checked before (adding and removing "!" operators in the if conditions). This seems to indicate that none of this code is covered by tests, not even manual ones. Do you see a chance to add some test cases for this?

**Comment by adimova** [comment]
This code is a result of a few changes that have not been tested manually in all situations and obviously there is no coverage from automated testing.
This exception was received as a result of changing the criterion for discovering which bundle provides resources and which not. Another change came from fix the fragments, etc. i will not list them all.
Obviously we need the automated testing to help us to find the regressions when making such changes.

**Comment by marinmitev** [comment]
There is a similar test public void 'assert that a rule switches from IDLE to NOT_INITIALIZED if a moduleHanlder disappears and back to IDLE if it appears again' ()
in the file
\smarthome\bundles\automation\org.eclipse.smarthome.automation.integration.test\src\test\groovy\org\eclipse\smarthome\automation\integration\test\AutomationIntegrationTest.groovy
which can be easily extended to check not only for rule state when bundle disappears but also to check for existing module types from ModuleTypeRegistry

File: bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java (hunk 1 above)
Comment (adimova) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java]: this situation may come up if the event for uninstalling bundle come until the event for installing the same bundle still processing - parsing the json

Comment (adimova) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java]: @kaikreuzer what you suggest, to log as debug or info, or to remove logging?

Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java]: I would vote for debug logging.

**Comment by adimova** [comment]
updated

**Comment by maggu2810** [comment]
So, wait for tests here or merge this one and add an issue to track that tests should be added?

**Comment by marinmitev** [comment]
I think it is better to separate the bug fix and the test code

**Comment by danchom** [comment]
I've tested manually Ani's fix and it works. I also think it is better to close this bug and add a new one for the tests.

**Comment by maggu2810** [comment]
Please create a new issue and link it here.

**Comment by maggu2810** [comment]
Thanks

**Pull request merged**